File: 30DaysOfCode/Day 11 - 2D Arrays.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int arr[][] = new int[6][6];
        for(int i=0; i < 6; i++){
            for(int j=0; j < 6; j++){
                arr[i][j] = in.nextInt();
            }
        }
       
       int res=0;
       int max = Integer.MIN_VALUE;
       for(int i=0;i<4;i++) {
    	   for(int j=0;j<4;j++) {    		   		
    		   		res = arr[i][j] +  arr[i][j+1]   + arr[i][j+2]+
    				   	  			   arr[i+1][j+1] +
    				   	  arr[i+2][j] +arr[i+2][j+1] + arr[i+2][j+2];
    		   		//System.out.println(res);
	    		   if(res>max)
	    			   max = res;
	    		    //System.out.println(max);
    	   }
       }
       System.out.println(max);
    }
}


Sample Input:

1 1 1 0 0 0
0 1 0 0 0 0
1 1 1 0 0 0
0 0 2 4 4 0
0 0 0 2 0 0
0 0 1 2 4 0

Sample Output:

19

File: 30DaysOfCode - Kotlin/Day_11_2d_arrays.kt
import java.util.*

fun main(args: Array<String>) {

    val input = Scanner(System.`in`)

    val array = Array(6) { IntArray(6) }
    for(i in 0 until 6) {
        for(j in 0 until  6){
            array[i][j] = input.nextInt()
        }
    }

    var max = Integer.MIN_VALUE
    var result: Int

    for(i in 0 until 4) {
        for(j in 0 until  4){
            result = array[i][j] + array[i][j+1] + array[i][j+2]+
                    array[i+1][j+1]+
                    array[i+2][j] + array[i+2][j+1] + array[i+2][j+2];
            if(result > max)
                max = result;
        }
    }

    println(max)
}
